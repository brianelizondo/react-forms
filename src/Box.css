.Box{
    margin-bottom: 20px;
}

.Box div{
    float:"left" !important;
}