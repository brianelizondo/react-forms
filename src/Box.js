import React, { useState } from "react";
import "./Box.css";

function Box({ color, width, height, deleteBox, id }){
    const handleClick = () => {
        deleteBox(id);
    }

    return (
        <div className="Box">
            <div style={{backgroundColor: color, width:`${width}px`, height:`${height}px`}}></div>
            <button onClick={handleClick}>X</button>
        </div>
    );
}

export default Box;