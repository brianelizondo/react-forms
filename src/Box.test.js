import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box";

// Smoke Tests
it("'Box' component renders without crashing", function() {
    render(<Box />);
});

// Snapshot Tests
it("'Box' matches snapshot", function() {
    const {asFragment} = render(<Box width={200} height={200} color="blue" id={123} key={123}/>);
    expect(asFragment()).toMatchSnapshot();
});