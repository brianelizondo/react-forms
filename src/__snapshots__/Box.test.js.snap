// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'Box' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      style="background-color: blue; width: 200px; height: 200px;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
